### PR TITLE
Feature/sync zoom and center

### PR DESCRIPTION
**Pull request by Gebuz**
Syncs zoom and centering (not the marker, but the center of the maps/images) when dragging the map or zooming in/out.

missing:
 - changing the main map resets the center to the marker
 - Should extent be removed from the images? risks empty maps (you can drag the images out of the frame). The alternative means the center wont always be synced when close to an edge because the images will get "pushed" to fit in the frame.

### CODE REVIEW

**Comment by Gebuz** [comment]
Switching view no longer recenters on the marker.
Extent stays.

File: src/components/viewport.js [hunk 1]
```diff
@@ -180,10 +180,13 @@ export class SkraaFotoViewport extends HTMLElement {
...
-    this.view.center = this.coord_image
-
-    // removes extent from the view
-    // this.view.extent = undefined
+    // this.view.center = this.coord_image
+    const center = store.state.view.center
+    if (center[0]) {
+      this.view.center = world2image(this.item, center[0], center[1], center[2])
+    } else {
+      this.view.center = this.coord_image
```
Comment (iamfrank) [inline src/components/viewport.js:188]: Er det så længere nødvendigt at checke for `this.coord_image` i starten af denne function?

**Comment by rasmustoender** [comment]
Feature works as supposed. 

"Should extent be removed from the images?"
- I think an allowed margin between image and frame could solve the issue on both sides. So that you can move out of the image boundary to a certain degree, and therefore keep the center as intended

**Review by rasmustoender — APPROVED** [review]

**Comment by Gebuz** [comment]
if we want a bigger extent, we can add this in `viewport.js`'s `updateMap()` function:

```
// increase extent in all directions by 20%
const i = 0.2
const ex = this.view.extent
this.view.extent = [ex[0] - i * ex[2], ex[1] - i * ex[3], (1 + i) * ex[2], (1 + i) * ex[3]]
```

**Pull request merged**